app/src/main/java/org/p2p/wallet/send/model/SendSolanaFee.kt

```diff
@@ -143,10 +151,13 @@ data class SendSolanaFee constructor(
                 if (desiredAmount != null) ReduceInputAmount(desiredAmount) else SwitchToSol
             }
             // if there is enough SPL token balance to cover amount and fee
-            !isSourceSol && sourceTokenTotal.isMoreThan(totalNeeded) ->
-                UpdateFeePayer
-            else ->
-                SwitchToSol
+            !isSourceSol && sourceTokenTotal.isMoreThan(totalNeeded) -> UpdateFeePayer
+            hasAlternativeFeePayerTokens -> {
+                alternativeFeePayerTokens.firstOrNull { it.totalInLamports.isMoreThan(totalNeeded) }
+                    ?.let { token -> FeePayerState.SwitchToSpl(token) }
+                    ?: SwitchToSol
```

Comment:
you don't need `firstOrNull` here because you already added the check above. 
You can use `hasAlternativeFeePayerTokens -> alternativeFeePayerTokens.first()`